com.reprezen.swagedit/src/com/reprezen/swagedit/model/AbstractNode.java

```diff
@@ -0,0 +1,150 @@
+import com.reprezen.swagedit.json.SchemaDefinitionProvider;
+
+public abstract class AbstractNode {
+
+    private static SchemaDefinitionProvider provider = new SchemaDefinitionProvider();
+
+    private final JsonPointer pointer;
+    private String property;
+    private AbstractNode parent;
+    private JsonType type;
+    private JsonNode schema;
+
+    public final Set<SchemaDefinition> definitions;
+
+    private JsonLocation location;
+
+    private JsonLocation start;
+
+    private JsonLocation end;
+
+    AbstractNode(AbstractNode parent, JsonPointer ptr) {
+        this(parent, ptr, null);
+    }
+
+    AbstractNode(AbstractNode parent, JsonPointer ptr, JsonLocation location) {
+        this.parent = parent;
+        this.pointer = ptr;
+        this.location = location;
+        this.definitions = provider.getDefinitions(pointer(ptr));
+        this.schema = definitions.isEmpty() ? null : Iterables.getFirst(definitions, null).definition;
+        this.type = schema != null ? JsonType.valueOf(this.schema) : null;
+    }
+
+    public AbstractNode get(int pos) {
+        return null;
+    }
+
+    public AbstractNode get(String property) {
+        return null;
+    }
+
+    public abstract boolean isObject();
+
+    public abstract boolean isArray();
+
+    public ObjectNode asObject() {
+        return (ObjectNode) this;
+    }
+
+    public ArrayNode asArray() {
+        return (ArrayNode) this;
+    }
+
+    public ValueNode asValue() {
+        return (ValueNode) this;
+    }
+
+    public JsonLocation getLocation() {
+        return location;
+    }
+
+    protected static String pointer(JsonPointer pointer) {
+        return pointer.toString().replaceAll("/", ":").replaceAll("~1", "/");
+    }
+
+    public JsonPointer getPointer() {
+        return pointer;
+    }
+
+    public JsonType getType() {
+        return type;
+    }
+
+    public JsonNode getSchema() {
+        return schema;
+    }
+
+    public Set<SchemaDefinition> getDefinitions() {
+        return definitions;
+    }
+
+    public AbstractNode getParent() {
+        return parent;
+    }
+
+    public String getProperty() {
+        return property;
+    }
+
+    public void setProperty(String name) {
+        this.property = name;
+    }
+
+    public Iterable<AbstractNode> elements() {
+        return Collections.emptyList();
+    }
+
+    public abstract String getText();
+
+    public Position getPosition(IDocument document) {
+        JsonLocation location = getStart() != null ? getStart() : getLocation();
+        if (location != null) {
+
+            int startLine = location.getLineNr() - 1;
+            int offset = 0;
+            try {
+                offset = document.getLineOffset(startLine) + location.getColumnNr()
+                        + Strings.nullToEmpty(getProperty()).length();
+            } catch (BadLocationException e) {
+                return new Position(0);
+            }
+
+            return new Position(Math.max(0, offset), 0);
```

Comment:
Because the length (the second parameter) is always null navigation did not work and Quick Outline was basically useless. I provided a simple fix in commit d90ea00390076f38d982d68df1adb37623698b2e which selects the entire first line of an element:
<img width="680" alt="screen shot 2016-08-11 at 4 21 53 pm" src="https://cloud.githubusercontent.com/assets/644582/17603566/c3f456b0-5fdf-11e6-96c2-9cc02497f238.png">
@ghillairet please review update my fix and update if needed.